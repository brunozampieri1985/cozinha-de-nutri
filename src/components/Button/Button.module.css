.btn {
    border-radius: 4px;
    border: 0;
    font-weight: bold;
    cursor: pointer;
}

.outline {
    color: var(--theme-600);
    background-color: var(--theme-100);
    border: 1px solid var(--theme-600);
    
}
.outline:hover {
    color: var(--theme-600);
    background-color: var(--theme-200);
    border: 1px solid var(--theme-600);
    box-shadow: 0px 0px 10px var(--theme-600);
}

.secondary {
    color: #fff;
    background-color: #777;
    border-color: #777;
}
.secondary:hover {
    color: #fff;
    background-color: #555;
    border-color: #555;
    box-shadow: 0px 0px 10px #555;
}

.primary {
    color: #fff;
    background-color: var(--theme-700);
    border: 1px solid var(--theme-700);
    
}
.primary:hover {
    color: #fff;
    background-color: var(--theme-800);
    border: 1px solid var(--theme-800);
    box-shadow: 0px 0px 10px var(--theme-700);
}

.small {
    padding: 5px 10px;
    font-size: 12px;
    line-height: 1.5;
}

.medium {
    padding: 10px 20px;
    font-size: 14px;
    line-height: 1.5;
}

.large {
    padding: 15px 30px;
    font-size: 16px;
    line-height: 1.5;
}