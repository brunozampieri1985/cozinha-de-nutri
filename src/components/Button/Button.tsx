import styles from './Button.module.css'

interface ButtonProps {
  children: React.ReactNode
  variant?: 'primary' | 'secondary' | 'outline'
  size?: 'small' | 'medium' | 'large'
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
  style?: React.CSSProperties
}

const Button: React.FC<ButtonProps> = ({
  children,
  variant,
  size,
  style,
  onClick,
}) => {
  return (
    <button
      className={`${styles.btn} ${
        styles[variant !== undefined ? variant : 'primary']
      } ${styles[size !== undefined ? size : 'medium']}`}
      style={style}
      onClick={onClick}>
      {children}
    </button>
  )
}

export default Button