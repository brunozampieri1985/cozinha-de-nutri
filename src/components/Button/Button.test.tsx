import { screen, render } from '@testing-library/react'
import Button from './Button'


describe('Button', () => {
    it('should render correctly', () => {
        const { container } = render(<Button>Hello</Button>)
        expect(container).toMatchSnapshot()
    })
    it('should render a primary button', () => {
        render(<Button variant='primary'>Hello</Button>)
        expect(screen.getByText('Hello')).toHaveClass('primary')
    })
    it('should render a secondary button', () => {
        render(<Button variant='secondary'>Hello</Button>)
        expect(screen.getByText('Hello')).toHaveClass('secondary')
    })
    it('should render a outline button', () => {
        render(<Button variant='outline'>Hello</Button>)
        expect(screen.getByText('Hello')).toHaveClass('outline')
    })
    it('should render a small button', () => {
        render(<Button size='small'>Hello</Button>)
        expect(screen.getByText('Hello')).toHaveClass('small')
    })
    it('should render a large button', () => {
        render(<Button size='large'>Hello</Button>)
        expect(screen.getByText('Hello')).toHaveClass('large')
    })
    it('should render a medium button', () => {
        render(<Button size='medium'>Hello</Button>)
        expect(screen.getByText('Hello')).toHaveClass('medium')
    })
    it('should have a click function', () => {
        const fn = jest.fn()
        render(<Button onClick={fn}>Hello</Button>)
        screen.getByText('Hello').click()
        expect(fn).toHaveBeenCalled()
    })
})
   