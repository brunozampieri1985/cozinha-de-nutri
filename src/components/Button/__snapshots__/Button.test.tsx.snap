// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<div>
  <button
    class="btn primary medium"
  >
    Hello
  </button>
</div>
`;
